static/app/views/insights/common/utils/useStarredSegment.tsx

```diff
@@ -58,7 +59,7 @@ export function useStarredSegment({initialIsStarred, projectSlug, segmentName}:
   const isPending = starTransactionResult.isPending || unstarTransactionResult.isPending;
 
   const toggleStarredTransaction = () => {
-    if (isPending) {
+    if (isPending || !project?.id) {
```

Comment:
🤔 in this state where there's no project ID, should it return a prop so the star component is disabled?